tools/ALARAJOYWrapper/njoy_tools.py

```diff
@@ -230,13 +266,25 @@ def run_njoy(element, A, matb):
         matb (int): Unique material ID for the material in the files.
     
     Returns:
-        gendf_path (str or None): File path to the newly created GENDF file.
+        save_dict['GENDF']['Save Path'] (str or None): File path to the newly 
+                                    created GENDF file.
                                     Returns None if NJOY runs unsuccessfuly.
         result.stderr (str or None): Output of NJOY error.
                                     Returns None if NJOY runs successfully. 
     """
 
-    gendf_path = None
+    file_metadata = [
+        ('GENDF', 'gendf_files', '.gendf', 31),
+        ('PENDF', 'pendf_files', '.pendf', 25)
+    ]
+    save_dict = {
+        name: {
+            'Specific Dir'              :           spec_dir,
+            'Extension'                 :                ext,
+            'Tape'                      :               tape
+        }
+        for name, spec_dir, ext, tape in file_metadata
+    }
```

Comment:
I wonder if it's simpler to just build the dictionary from scratch rather than expanding the tuples into a dictionary:
```
file_metadata = {
    'GENDF' : { 'Specific Dir' : 'gendf_files', 'Extension' : '.gendf', 'Tape' : 31 }
    'PENDF' : { 'Specific Dir' : 'pendf_files', 'Extension' : '.pendf', 'Tape' : 25 }
}
```
I might even advocate for shorter keys in the nested dictionaries: `dir', `ext`, `tape`